### ISSUE TITLE
[MCHECKSTYLE-305] Allow simultaneous use of <configLocation> and <checkstyleRules> as override

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-checkstyle-plugin#602]
**[Pierre Maréchal](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=smokeshield)** opened **[MCHECKSTYLE-305](https://issues.apache.org/jira/browse/MCHECKSTYLE-305?redirect=false)** and commented

It would be a nice addition to somehow allow the use of built-in config files, yet being able to override some of the settings without adding the whole configuration inline.

My use case is the following: We use google_checks.xml as a baseline, but want to override the builtin \<module name="Indentation"/> with its stock config. At the moment, I must input the whole google_checks.xml as inline configuration, or revert to bulding a specific artifact to include my slightly modified configuration as a dependency (I use multi module).

Additionally, due to MCHECKSTYLE-304  I cannot even input the google_checks as is.

In summary, it would allow tweaking the built-in configuration, without losing upstream evolution.

---

**Affects:** 2.16

4 votes, 4 watchers
